### PR TITLE
Add BottomNavigationBarType.shifting sample #72936

### PR DESCRIPTION
**Pull request by emem365**
## Description

 To better enable users to understand the usage of `BottomNavigationBarType.shifting` as mentioned in #72936, this PR adds a Sample for BottomNavigationBar using four BottomNavigationBarItem widgets which makes the BottomNavigationBar widget to use BottomNavigationBarType.shifting by default

## Related Issues

#72936

## Tests

This PR adds changes to Documentation only and thus requires no tests

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by emem365** [comment]
@shihaohong , this is a PR for the issue #72936. Please take a look. Also, the PR uses an asset that needs to be generated and added to the assets-for-api-docs. Should I try to do that and create a PR for that as well?

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 1]
```diff
@@ -19,7 +19,7 @@ import 'theme.dart';
 import 'tooltip.dart';
 
 /// Defines the layout and behavior of a [BottomNavigationBar].
-///
+/// For a sample on how to use these, please see [BottomNavigationBar].
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart:22]: We usually leave a line between the leading sentence and any descriptions that follow.

We also usually leave a line between the See also section and the description.

```suggestion
///
/// For a sample on how to use these, please see [BottomNavigationBar].
///
```

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 2]
```diff
@@ -133,6 +134,89 @@ enum BottomNavigationBarType {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows a [BottomNavigationBar] as it is used within a [Scaffold]
+/// widget. The [BottomNavigationBar] has four [BottomNavigationBarItem]
+/// widgets, which means it defaults to [BottomNavigationBarType.shifting], and
+/// the [currentIndex] is set to index 0. The selected item is amber.
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart:141]:
```suggestion
/// the [currentIndex] is set to index 0. The selected item is amber in color.
```

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 3]
```diff
@@ -133,6 +134,89 @@ enum BottomNavigationBarType {
...
+/// With each [BottomNavigationBarItem] widget, backgroundColor property is
+/// also defined, which changes the background color of [BottomNavigationBar],
+/// according to the selected item. The `_onItemTapped` function changes the
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart:144]:
```suggestion
/// when that item is selected. The `_onItemTapped` function changes the
```

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 4]
```diff
@@ -133,6 +134,89 @@ enum BottomNavigationBarType {
...
+///            label: 'Settings',
+///            backgroundColor: Colors.pink,
+///          ),
+///        ],
+///        currentIndex: _selectedIndex,
+///        selectedItemColor: Colors.amber[800],
+///        onTap: _onItemTapped,
+///      ),
+///    );
+///  }
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart:217]: nit: This entire block is one space too indented to the right.

**Review by shihaohong — APPROVED** [review]
Some nits, otherwise LGTM!

**Comment by emem365** [comment]
@shihaohong I have committed the changes you suggested and also fixed the indentation.

One thing that still needs to be done is to add an animated gif or a screenshot to the sample like the [BottomNavigationBarType.fixed] has an image. Should I try to open a PR to [assets-for-api-docs repo](https://github.com/flutter/assets-for-api-docs)?

**Comment by shihaohong** [comment]
To be honest, I think the image is unnecessary since DartPad pulls up a live sample app, making the image redundant. If anything, the images should be removed from the samples where DartPad sample code exists. 

On the other hand, maybe the right thing to do is the opposite -- remove the existing image for the `fixed` sample.

**Comment by emem365** [comment]
@shihaohong That makes perfect sense! Made a new commit removing the image from the doc.

**Pull request merged**